src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -192,9 +193,12 @@ const IndexPatternEditorFlyoutContentComponent = ({
 
   const loadTimestampFieldOptions = useCallback(
     async (query: string) => {
+      const currentLoadingTimestampFieldsIdx = ++currentLoadingTimestampFieldsRef.current;
       let timestampOptions: TimestampOption[] = [];
       const isValidResult =
-        !existingIndexPatterns.includes(query) && matchedIndices.exactMatchedIndices.length > 0;
+        !existingIndexPatterns.includes(query) &&
+        matchedIndices.exactMatchedIndices.length > 0 &&
+        !isLoadingMatchedIndices;
```

Comment:
The issue was happening when `query` was updated but we still didn't get the matched indices to result for the latest query: 

we got into a state where `query` is actually doesn't match any indices but `matchedIndices.exactMatchedIndices.length > 0` because it keeps results for the previous queries.